inference-engine/src/transformations/src/transformations/op_conversions/convert_interpolate1_to_interpolate4.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/op_conversions/convert_interpolate1_to_interpolate4.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/opsets/opset4.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ConvertInterpolate1ToInterpolate4, "ConvertInterpolate1ToInterpolate4", 0);
+
+ngraph::pass::ConvertInterpolate1ToInterpolate4::ConvertInterpolate1ToInterpolate4() {
+    auto interpolate1 = ngraph::pattern::wrap_type<ngraph::opset1::Interpolate>();
+
+    ngraph::matcher_pass_callback callback = [this](pattern::Matcher& m) {
+        auto interpolate1 = std::dynamic_pointer_cast<ngraph::opset1::Interpolate>(m.get_match_root());
+        if (!interpolate1)
+            return false;
+
+        auto data_node = interpolate1->input_value(0);
+        auto out_shape_node = std::dynamic_pointer_cast<ngraph::opset1::Constant>(interpolate1->input_value(1).get_node_shared_ptr());
+        auto interpolate_attrs = interpolate1->get_attrs();
+        auto input_shape = data_node.get_shape();
```

Comment:
At first shape can be dynamic so you need to check it. But you use only shape rank so it will be enough to get partial shape rank length. 
```suggestion
        auto input_shape_rank = interpolate1->input(0).get_partial_shape().rank().get_length();
```
And also you need to check that rank is not static before calling `get_length` method. And this can be done via pattern predicate:
```cpp
 auto interpolate1 = ngraph::pattern::wrap_type<ngraph::opset1::Interpolate>({pattern::any_input(pattern::has_static_rank()), pattern::any_input());
```

Reply:
fixed. thank you!